validation/2.0/modules/ROOT/pages/validation-documentation.adoc

```diff
@@ -0,0 +1,785 @@
+
+Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#anypoint-connectors[Select]
+
+Validation Module v2.0
+
+The Validation module can verify that the contents of a message in a Mule flow match specific criteria. The module provides explicit messages about the cause of an exception in a flow. Based on the specific data you are validating, you can customize the exception message that is displayed in the logs.
+
+A module that performs data validations. If the validation fails, an error is thrown. The error type might vary but it will always be a child of VALIDATION:VALIDATION.
+
+If a message does not meet the defined validation criteria, the validation fails and returns a `VALIDATION` error (such as `VALIDATION:INVALID_URL`), which is a `ValidationException`. By default, this message includes a meaningful message that you can customize.
+
+Release Notes: xref:release-notes::mule-runtime/module-validation.adoc[Validation Module Release Notes]
+
+
+== Configurations
+---
+[[Config]]
+=== Config
+
+
+Default configuration
+
+
+==== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+|Name | String | Name for this configuration. Connectors reference the configuration with this name. | | x
+| I18n a| <<I18n>> |  Allows to configure I18n for the standard error messages |  |
+| Name a| String |  ID used to reference this configuration. |  | x
+| Expiration Policy a| <<ExpirationPolicy>> |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before Mule considers it eligible for expiration.   |  |
+|===
+
+
+== Supported Operations
+* <<IsAllowedIp>>
+* <<IsBlankString>>
+* <<IsElapsed>>
+* <<IsEmail>>
+* <<IsEmptyCollection>>
+* <<IsFalse>>
+* <<IsIp>>
+* <<IsNotBlankString>>
+* <<IsNotDeniedIp>>
+* <<IsNotElapsed>>
+* <<IsNotEmptyCollection>>
+* <<IsNotNull>>
+* <<IsNull>>
+* <<IsNumber>>
+* <<IsTime>>
+* <<IsTrue>>
+* <<IsUrl>>
+* <<MatchesRegex>>
+* <<ValidateSize>>
+
+
+
+== Operations
+
+[[IsAllowedIp]]
+== Is Allowed Ip
+`<validation:is-allowed-ip>`
+
+
+Validates that a IP address is present in the IP list.
+
+
+=== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | Name of the configuration to use. | | x
+| Ip Address a| String | Address to validate. |  | x
+| Allow List a| <<IpFilterList>> |  List of allowed addresses. |  | x
+| Config Ref a| ConfigurationProvider |  Name of the configuration to use to execute this component. |  |
+| Message a| String |  SSpecifies the message to notify the user if the validation fails. The message is marked as not dynamic to allow eager evaluation of the expression in case the validation is successful and the message is not required. Components consuming this value check if this is an expression and evaluate it in case the validation failed. |  |
```

Comment:
```suggestion
| Message a| String |  Specifies the message to notify the user if the validation fails. The message is marked as not dynamic to allow eager evaluation of the expression in case the validation is successful and the message is not required. Components consuming this value check if this is an expression and evaluate it in case the validation failed. |  |
```